TMS.Net07.Homework.Calculator/Beginner/Program.cs

```diff
@@ -0,0 +1,175 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Beginner
+{
+    class Program
+    {
+        // National Bank of The Republic of Belarus (07.02.2021)
+        const double BYNtoUSD = 0.3805;
+        const double BYNtoEUR = 0.318;
+        const double BYNtoRUB = 28.5527;
+        const double USDtoBYN = 2.6282;
+        const double USDtoEUR = 0.8357;
+        const double USDtoRUB = 75.0421;
+        const double EURtoBYN = 3.1448;
+        const double EURtoUSD = 1.1966;
+        const double EURtoRUB = 89.7924;
+        const double RUBtoBYN = 0.035023;
+        const double RUBtoUSD = 0.013326;
+        const double RUBtoEUR = 0.011137;
+
+        static void Main(string[] args)
+        {
+            System.Threading.Thread.CurrentThread.CurrentCulture = new System.Globalization.CultureInfo("en-US");
+            while (true)
+            {
+                Console.Clear();
+                Console.WriteLine("Supported currencies: BYN, USD, EUR, RUB", Environment.NewLine);
+                string source;
+                Console.WriteLine($"{Environment.NewLine}Input source currency:");
+                Console.Write("-> ");
+                while (true)
+                {
+                    source = Console.ReadLine().ToUpper();
+                    if (source != "BYN" && source != "USD" && source != "EUR" && source != "RUB")
+                    {
+                        Console.WriteLine($"{Environment.NewLine}Please input supported currency:");
+                        Console.Write("-> ");
+                        continue;
+                    }
+                    break;
+                }
+
+                string target;
+                Console.WriteLine($"{Environment.NewLine}Input target currency:");
+                Console.Write("-> ");
+                while (true)
+                {
+                    target = Console.ReadLine().ToUpper();
+                    if (target != "BYN" && target != "USD" && target != "EUR" && target != "RUB")
+                    {
+                        Console.WriteLine($"{Environment.NewLine}Please input supported currency:");
+                        Console.Write("-> ");
+                        continue;
+                    }
+                    if (target == source)
+                    {
+                        Console.WriteLine($"{Environment.NewLine}Please input another currency:");
+                        Console.Write("-> ");
+                        continue;
+                    }
+                    break;
+                }
+
+                double amount;
+                Console.WriteLine($"{Environment.NewLine}Input amount:");
+                Console.Write("-> ");
+                while (true)
+                {
+                    try
+                    {
+                        amount = Convert.ToDouble(Console.ReadLine());
```

Comment:
Use `double.TryParse` instead. There are 2 reasons for that:
- `Convert.ToDouble` is slower, that `double.TryParse`
- throwing and catching exceptions is a much more resource consuming operation, than just checking the TryParse boolean result.

Reply:
DONE

Thank you. In all three tasks `try{} catch{}` blocks were replaced with manual checks. All `Convert.ToDouble()` were replaced with `Double.TryParse()`.